File: RepositoryPattern/SampleApplicationForGM/app/src/main/java/com/example/sampleapplicationforgm/viewmodel/DetailActivityViewModel.java
package com.example.sampleapplicationforgm.viewmodel;

import javax.inject.Inject;

import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.OnLifecycleEvent;
import androidx.lifecycle.ViewModel;

public class DetailActivityViewModel extends ViewModel implements LifecycleObserver {

    @Inject
    public DetailActivityViewModel() {
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    public void onCreate() {
        System.out.println("DetailActivityViewModel onCreate called ");
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    public void onResume() {
        System.out.println("DetailActivityViewModel onResume called ");
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    public void onStart() {
        System.out.println("DetailActivityViewModel onStart called ");
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    public void onStop() {
        System.out.println("DetailActivityViewModel onStop called ");
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    public void onPause() {
        System.out.println("DetailActivityViewModel onPause called ");
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
    public void onDestroy() {
        System.out.println("DetailActivityViewModel onDestroy called ");
    }

    @Override
    protected void onCleared() {
        super.onCleared();
    }
}

File: RepositoryPattern/SampleApplicationForGM/app/src/main/java/com/example/sampleapplicationforgm/viewmodel/DetailActivityViewModel.kt
package com.example.sampleapplicationforgm.viewmodel

import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.OnLifecycleEvent
import androidx.lifecycle.ViewModel
import javax.inject.Inject

class DetailActivityViewModel @Inject constructor() : ViewModel(), LifecycleObserver {
    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    fun onCreate() {
        println("DetailActivityViewModel onCreate called ")
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
    fun onResume() {
        println("DetailActivityViewModel onResume called ")
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_START)
    fun onStart() {
        println("DetailActivityViewModel onStart called ")
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    fun onStop() {
        println("DetailActivityViewModel onStop called ")
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    fun onPause() {
        println("DetailActivityViewModel onPause called ")
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
    fun onDestroy() {
        println("DetailActivityViewModel onDestroy called ")
    }

    override fun onCleared() {
        super.onCleared()
    }
}